### PR TITLE
ws: migrate to an RPC arch

### PR DESCRIPTION
**Pull request by sanderpick**
This migrates the WS server to an RPC style where incoming requests are mapped to responses via a request ID, i.e., https://github.com/textileio/js-threads-client/blob/master/src/index.ts. 

I've stubbed out the client methods which mirror `Threadservice` but will need to do some additional testing. That being said, I'd still take some eyes on this in the meantime.

### CODE REVIEW

**Comment by sanderpick** [comment]
This is ready for some eyes.

File: cbor/event.go [hunk 1]
```diff
@@ -47,14 +46,14 @@ func NewEvent(ctx context.Context, dag format.DAGService, body format.Node, sett
 		return nil, err
 	}
 	eventHeader := &eventHeader{
-		Time: settings.Time.Unix(),
+		Time: time.Now().Unix(),
```
Comment (carsonfarmer) [inline cbor/event.go]: Should we think about getting an HCL implementation in here soon? Something like https://github.com/lafikl/hlc looks very simple?

File: ws/client.go [hunk 2]
```diff
@@ -106,94 +261,49 @@ func (c *Client) readPump() {
...
+			switch caller.Method {
+			case "addThread":
+				result, err = c.addThread(ctx, caller.Args...)
+			case "pullThread":
+				err = fmt.Errorf("todo")
+			case "deleteThread":
+				err = fmt.Errorf("todo")
+			case "addFollower":
+				err = fmt.Errorf("todo")
+			case "addRecord":
```
Comment (carsonfarmer) [inline ws/client.go]: Why not these ones yet? Specific reason, or just not there yet?

File: ws/hub.go [hunk 3]
```diff
@@ -67,3 +66,11 @@ func (h *Hub) run() {
 		}
 	}
 }
+
+// encodeRecord base64 encodes a thread record.
+func encodeRecord(rec thread.Record) []byte {
```
Comment (carsonfarmer) [inline ws/hub.go]: Nice

File: ws/client.go (hunk 2 above)
Comment (sanderpick) [reply ws/client.go]: Oops! I forgot to enable the new methods... added.

File: cbor/event.go (hunk 1 above)
Comment (sanderpick) [reply cbor/event.go]: Nice—issue added: https://github.com/textileio/go-textile-threads/issues/76

**Comment by carsonfarmer** [comment]
Nice, re-approved 👍.

**Pull request merged**